lib/src/api/opt/query.rs

```diff
@@ -258,9 +266,9 @@ where
 {
 	fn query_result(self, QueryResponse(map): &mut QueryResponse) -> Result<Option<T>> {
 		let (index, key) = self;
-		let vec = match map.get_mut(&index) {
+		let mut value: &mut Value = match map.get_mut(&index) {
```

Comment:
```suggestion
		let value = match map.get_mut(&index) {
```

Not sure about this one without actually running it but this should be fine because `get_mut` returns a mutable reference. The result should be concrete too so there should be no need for a type hint here.